Sources/Shared/TCPSocket.cpp

```diff
@@ -42,7 +42,9 @@ int TCPSocket::send(const std::string &data) {
   return ::send(this->socketDescriptor, data.c_str(), data.size(), 0);
 }
 
-int TCPSocket::receive(std::string &data) {
+int TCPSocket::receive(std::string &data, int bytes) {
+  char buffer[bytes];
```

Comment:
1.This is not implemented yet fully.
2. Why is that? This buffer is only living in this place, and nothing else. We need this memory only while this function is running.
With comments please write a little bit of description, so that we could have a discussion here.
Thanks,